openhexa/sdk/pipelines/runtime.py

```diff
@@ -19,6 +22,47 @@ def import_pipeline(pipeline_dir_path: str):
     return pipeline
 
 
+def get_pipeline_specs(pipeline_dir_path) -> typing.Tuple[PipelineSpecs, typing.List[PipelineParameterSpecs]]:
+    pipeline_dir = os.path.abspath(pipeline_dir_path)
+    sys.path.append(pipeline_dir)
+    pipeline_node = None
+
+    with open(pipeline_dir_path / Path("pipeline.py")) as f:
+        tree = ast.parse(f.read())
+        for node in ast.walk(tree):
+            if isinstance(node, ast.FunctionDef) and [
+                decorator for decorator in node.decorator_list if decorator.func and decorator.func.id == "pipeline"
```

Comment:
Is there any way to make sure that the decorator is the actual `openhexa.sdk.pipeline` function - and not any function called `pipeline`?

Reply:
I guess you can yes.

The list comprehension is weird.

```python
if isinstance(node, ast.FunctionDef) and any([dec.func.id == "pipeline" for dec in node.decorator_list])
```

I would anyway move that check in a separate function and not use the id of the function but check if it comes from the module openhexa.sdk

Reply:
Based on my current research no, the ast module doesn't give that type of information, but I'll continue to look ak it.